File: appium/pages/settings_page.py
from pages.base_page import BasePage


class SettingsPage(BasePage):
    def assert_loaded(self) -> None:
        assert self.has_text("Settings") or self.has_text("User Profile")
        assert self.has_text("LOGOUT")

    def logout(self) -> None:
        self.tap_text("LOGOUT")
        # Confirm dialog
        self.tap_text("Logout")

File: selenium/pages/settingsPage.js
const { BasePage } = require("./basePage");

class SettingsPage extends BasePage {
  async assertLoaded() {
    const hasSettings = await this.hasText("Settings");
    const hasProfile = await this.hasText("User Profile");
    if (!hasSettings && !hasProfile) {
      throw new Error("Settings screen not loaded");
    }
    await this.assertHasText("LOGOUT");
  }

  async logout() {
    await this.tapText("LOGOUT");
    await this.tapText("Logout");
  }
}

module.exports = { SettingsPage };
